revolv/static/js/base/accounting.js

```diff
@@ -0,0 +1,388 @@
+	}
+
+	$('.cash-in-adjustment-total').append('<td class="added">' + total + "</td>");
+}
+
+/**
+ * Places a cash out AdminAdjustment element on the page.
+ * @param {Object} data_to_insert - A dictionary with values to insert.
+ * @param {String} time - The time period in which this was inserted.
+ */
+var allCashOutTransactions = {};
+function placeCashOutAdjustmentElement(data_to_insert, time) {
+
+	var total = data_to_insert['total'];
+	var transactions = data_to_insert['transactions'];
+
+	var allTransactionsKeys = Object.keys(allCashOutTransactions);
+	for (var i = 0; i < allTransactionsKeys.length; i++) {
+		allCashOutTransactions[allTransactionsKeys[i]] = 0;
+	}
+	//insert all the transactions
+	var transactionKeys = Object.keys(transactions)
+	for (var i = 0; i < transactionKeys.length; i++) {
+		var transactionName = transactionKeys[i];
+		var transactionAmount = transactions[transactionName];
+		var cleanedTransactionName = cleanUpString(transactionName);
+		allCashOutTransactions[cleanedTransactionName] = transactionAmount;
+	}
+	
+	allTransactionsKeys = Object.keys(allCashOutTransactions);
+	for (var i = 0; i < allTransactionsKeys.length; i++) {
+		var name = allTransactionsKeys[i];
+		var amount = allCashOutTransactions[name];
+		$('.transaction').filter('.' + name).append('<td class="added">' + amount + "</td>");
+	}
+
+	$('.cash-out-adjustment-total').append('<td class="added">' + total + "</td>");
+}
+
+/**
+ * Iterates through the data and inserts values onto the page.
+ * @param {Object} data - A dictionary of all the data
+ * @param {String} cash_in_or_cash_out - 'cash_in' for cash in, and 'cash_out' for cash out.
+ */
+function placeElements(data, cash_in_or_cash_out, subsection, insert_func) {
+	if (subsection == 'adjustments') {
+		var super_dict = data[cash_in_or_cash_out][subsection]
+		var time_keys = Object.keys(super_dict);
+		time_keys.sort();
+		for (var i = 0; i < time_keys.length; i++) {
+			var key = time_keys[i];
+			var transactionsAndTotal = super_dict[key];
+			insert_func(transactionsAndTotal, key);
+		}
+	} else {
+		var super_dict = data[cash_in_or_cash_out][subsection]
+		var time_keys = Object.keys(super_dict);
+		time_keys.sort();
+		for (var i = 0; i < time_keys.length; i++) {
+			var key = time_keys[i];
+			var project_super = super_dict[key];
+			var project_keys = Object.keys(project_super);
+			for (var j = 0; j < project_keys.length; j++) {
+				var pk = project_keys[j];
+				var data_to_insert = project_super[pk];
+				insert_func(data_to_insert, pk, key);
+			}
+		}
+	}
+}
+
+/**
+ * Sets the overal totals and the cash flow values.
+ * @param {Object} data - A dictionary with values to insert.
+ */
+function setTotalsAndCashFlow(data) {
+	//create the total row
+	//iterate through and create all the cash positions at each time
+	$('.cash-in-body').append('<tr class="added level-0 total-cash-in"><td class="level-0 added">Total Cash In</td><tr>');
```

Comment:
It looks like you probably don't need to append these, you can just render them by default and fill in their contents?